lib/oelite/cmd/bake.py

```diff
@@ -23,8 +28,46 @@ def parse_args(options, args):
     else:
         logging.getLogger().setLevel(logging.INFO)
 
+def os_check(config):
+    distro, release = setup.get_host_distro()
+    if distro and release:
+        setup_file, supported = setup.get_setup_file(config, distro, release)
+        if supported:
+            return True
+
+    if distro and release:
+        print('WARNING: Host distribution (%s %s) has not been validated'%(distro, release))
+    else:
+        print("WARNING: Cannot determine host distribution")
+
+    response = 'n'
+
+    if os.isatty(sys.stdin.fileno()):
+        while True:
+            try:
+                response = raw_input(
+                    "Do you want to continue [Y/n]? ")
+            except KeyboardInterrupt:
+                print
+                response = 'n'
+            response = response.lower()
+            if response == '':
+                response = 'y'
+            if response in ('y', 'n'):
+                print
+                break
+
+    if response == 'n':
+        print "Maybe another time"
+        return False
+    return True
```

Comment:
isn't this a duplication of the other place where we interactively ask if we should continue?

Let us not do that, we already have a mess of duplication and responsibility in the code. Please refactor this out to a place where it can be used by commands that need it.